library/cohesity_physical_server_job.py

```diff
@@ -0,0 +1,426 @@
+                    msg="The Protection job run is stopped",
+                )
+                module.exit_json(**result)
+            time.sleep(SLEEP_TIME_SECONDS)
+            wait_minutes = wait_minutes - 1
+        result = dict(
+            changed=True,
+            msg="Request to stop the protection job is made. Timed out while checking if the job is stopped"
+        )
+        module.exit_json(**result)
+    except APIException as ex:
+        raise__cohesity_exception__handler(str(
+            json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def create_physical_server_job(module):
+    '''
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    try:
+        request_body = ProtectionJobRequestBody()
+        request_body.name = module.params.get('job_name')
+        request_body.policy_id = get_policy_id(module)
+        request_body.view_box_id = get_view_box_id(module)
+        request_body.timezone = module.params.get('timezone')
+        if module.params.get('description'):
+            request_body.description = module.params.get('description')
+        if module.params.get('start_time'):
+            start_time = module.params.get('start_time').split(':')
+            if len(start_time) != 2:
+                raise__cohesity_exception__handler(
+                    "Please make sure the start time is in specified format", module)
+            time_of_day = TimeOfDay()
+            time_of_day.hour = int(start_time[0])
+            time_of_day.minute = int(start_time[1])
+            request_body.start_time = time_of_day
+        request_body.parent_source_id =\
+            get_physical_server_parent_id(module)
+        request_body.source_ids = get_physical_server_source_ids(module)
+        request_body.environment = 'kPhysical'
+        protection_job = cohesity_client.protection_jobs.create_protection_job(request_body)
+        return protection_job
+    except APIException as ex:
+        raise__cohesity_exception__handler(str(
+            json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def delete_physical_server_job(module, job_id):
+    '''
+    :param module: object that holds parameters passed to the module
+    :param job_id: protection job id
+    :return:
+    '''
+    try:
+        delete_job_params = DeleteProtectionJobParam()
+        delete_job_params.delete_snapshots = module.params.get('delete_snapshots')
+        cohesity_client.protection_jobs.delete_protection_job(job_id, body=delete_job_params)
+    except APIException as ex:
+        raise__cohesity_exception__handler(str(
+            json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def get_cohesity_client(module):
```

Comment:
can this be abstracted in util library, I see this being repeated in multiple files.

Reply:
ok